### PR TITLE
SBL - Measured boot support in UEFI payload

### PR DESCRIPTION
**Pull request by lsubash**
Added support for TPM event log info HOB and updated Loader FSP info.

### CODE REVIEW

File: BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c [hunk 1]
```diff
@@ -382,4 +382,51 @@ TpmLogLocalityEvent (
...
+  UINT32                Laml;            //LogAreaMinimumLength
+  UINT32                EventSize;
+  TCG_PCR_EVENT2_HDR    *EmptySlot;
+  TCG_PCR_EVENT_HDR     *FirstEvent;
+
+  GetTCGLasa (&Lasa, &Laml);
+
+  if (Lasa == 0 || Laml == 0 ) {
+    DEBUG ((DEBUG_WARN, "Unable to get log area for TCG 2.0 format events !!\n"));
+    *Tcg2Lasa = 0;
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c]: Is this function an external interface ?  If so, we need to check if Tcg2Lasa/Tcg2LasaEven is NULL.

File: BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c [hunk 2]
```diff
@@ -382,4 +382,51 @@ TpmLogLocalityEvent (
...
+  Get the TCG log Lasa and LasaEven info.
+
+  @param[out] Tcg2Lasa  Pointer to the Log area start address .
+  @param[out] Tcg2LasaEven  Pointer to the Last event logged.
+
+**/
+VOID
+GetTCGLogInfo (
+  OUT  UINT32  *Tcg2Lasa,
+  OUT  UINT32  *Tcg2LasaEven
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c]: Should we use Tcg2LasaLastEvent instead according to the function comments ?

File: BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c (hunk 1 above)
Comment (lsubash) [reply BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c]: Yes, its an external interface. I would add NULL check for Tcg2Lasa and Tcg2LasaEven.

File: BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c (hunk 2 above)
Comment (lsubash) [reply BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c]: Yeah.. I could change to Tcg2LasaLastEvent instead of Tcg2LasaEven.

**Comment by lsubash** [comment]
Thanks Sachin. I have added in UEFI payload functionality to retrieve TPM Event log info by accessing TPM2 ACPI table.  Able to get from events logged in Slim boot loader.

TPM Event log HOB need not be added to SBL.

File: BootloaderCorePkg/Stage2/Stage2Support.c [hunk 3]
```diff
@@ -390,6 +390,9 @@ BuildBaseInfoHob (
   if (LoaderFspInfo != NULL) {
     LoaderFspInfo->FspsBase   = PCD_GET32_WITH_ADJUST (PcdFSPSBase);
     LoaderFspInfo->FspHobList = LdrGlobal->FspHobList;
+    LoaderFspInfo->VerifiedBoot = FeaturePcdGet (PcdVerifiedBootEnabled);
+    LoaderFspInfo->MeasuredBoot = FeaturePcdGet (PcdMeasuredBootEnabled);
```
Comment (gdong1) [inline BootloaderCorePkg/Stage2/Stage2Support.c]: we could not just use these

















Even we build Verified boot and Measured boot (FeaturePcds are set to true in SBL), but the boot guard profile could disable them. should we use LOADER_PLATFORM_INFO->SecurityFlags instead?

Comment (lsubash) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: Sure, I would use interface LOADER_PLATFORM_INFO->SecurityFlags to update the security features.

**Comment by lsubash** [comment]
- Alternate implementation is done in UEFI payload to get TCG Event log info by reading ACPI tables.
- using the interface LOADER_PLATFORM_INFO to update security flags.

These patches would create an new HOB's which are no longer required.

**Pull request closed without merging**